airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -958,3 +958,96 @@ def execute(self, context: Context):
+    :param time_limit: The maximum amount of time in seconds to spend training the model(s).
+    :param autodeploy_endpoint_name: If specified, the best model will be deployed to an endpoint with
+        that name. No deployment made otherwise.
+    :param extras: Use this dictionary to set any variable input variable for job creation that is not
+        offered through the parameters of this function. The format is described in:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_auto_ml_job
+    :param wait_for_completion: Whether to wait for the job to finish before returning. Defaults to True.
+    :param check_interval: Interval in seconds between 2 status checks when waiting for completion.
+
+    :returns: Only if waiting for completion, a dictionary detailing the best model. The structure is that of
+        the "BestCandidate" key in:
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.describe_auto_ml_job
+    """
+
+    template_fields: Sequence[str] = (
+        "job_name",
+        "s3_input",
+        "target_attribute",
+        "s3_output",
+        "role_arn",
+        "compressed_input",
+        "time_limit",
+        "autodeploy_endpoint_name",
+        "extras",
+        "wait_for_completion",
+        "check_interval",
+    )
+
+    def __init__(
+        self,
+        *,
+        job_name: str,
+        s3_input: str,
+        target_attribute: str,
+        s3_output: str,
+        role_arn: str,
+        compressed_input: bool = False,
+        time_limit: int | None = None,
+        autodeploy_endpoint_name: str | None = None,
+        extras: dict | None = None,
+        wait_for_completion: bool = True,
+        check_interval: int = 30,
+        aws_conn_id: str = DEFAULT_CONN_ID,
+        config: dict | None = None,
+        **kwargs,
+    ):
+        super().__init__(config=config or {}, aws_conn_id=aws_conn_id, **kwargs)
+        self.job_name = job_name
+        self.s3_input = s3_input
+        self.target_attribute = target_attribute
+        self.s3_output = s3_output
+        self.role_arn = role_arn
+        self.compressed_input = compressed_input
+        self.time_limit = time_limit
+        self.autodeploy_endpoint_name = autodeploy_endpoint_name
+        self.extras = extras
+        self.wait_for_completion = wait_for_completion
+        self.check_interval = check_interval
+
+    def execute(self, context: Context) -> dict | None:
+        best = self.hook.create_auto_ml_job(
+            self.job_name,
+            self.s3_input,
+            self.target_attribute,
+            self.s3_output,
+            self.role_arn,
+            self.compressed_input,
+            self.time_limit,
+            self.autodeploy_endpoint_name,
+            self.extras,
+            self.wait_for_completion,
+            self.check_interval,
+        )
+        return best
```

Comment:
lets make sure best is not empty and the `self.do_xcom_push` is set

Reply:
well, best is going to be empty if we don't wait for completion. If we fire-and-forget, we don't really have anything to return. Do you think it would be a problem ?
It's documented in the docstring that this function returns something only if waiting for completion.

Reply:
operators return value is pushed to Xcom.
if best is empty then there is nothing to push thus the combination of `self.do_xcom_push` true + `best` not empty is when we want to return the value of `best`.

Reply:
Maybe I'm misunderstanding your point here, but `do_xcom_push` doesn't need to be checked here because it's handled once for everyone here, if I read it correctly:
https://github.com/apache/airflow/blob/7b5b3470c13dedbd699e0335548bdba7083cb29e/airflow/models/taskinstance.py#L1585-L1588
(It also makes sense that this piece of code wouldn't be repeated in every operator)

Also, `best` is coming from the hook, here
https://github.com/apache/airflow/blob/2a6fb076262cbba4c15eec9495e68f482d071247/airflow/providers/amazon/aws/hooks/sagemaker.py#L1245-L1247
So it should never be an empty dictionary. It's either something or None. I cannot _not_ return because it makes the static checker angry, but returning None is effectively the same.

I can add a check here that best is `!= {}` if that's what you meant ?